cleanlab/datalab/datalab.py

```diff
@@ -339,14 +342,25 @@ def report(
             print("Please specify some `issue_types` in datalab.find_issues() to see a report.\n")
             return
 
-        reporter = report_factory(self._imagelab)(
-            data_issues=self.data_issues,
-            task=self.task,
-            verbosity=verbosity,
-            include_description=include_description,
-            show_summary_score=show_summary_score,
-            imagelab=self._imagelab,
-        )
+        # TODO:refactor later
+        if self._imagelab:
+            reporter = ImagelabReporterAdapter(
+                data_issues=self.data_issues,
+                verbosity=verbosity,
+                include_description=include_description,
+                show_summary_score=show_summary_score,
+                imagelab=self._imagelab,
+            )
+        else:
+            reporter = Reporter(
+                data_issues=self.data_issues,
+                task=self.task,
+                verbosity=verbosity,
+                include_description=include_description,
+                show_summary_score=show_summary_score,
+                imagelab=self._imagelab,
+            )
+
```

Comment:
It's okay for you to update the factory so that it won't have to pass the task to `ImagelabReporterAdapter`.

Let the factory be responsible for deciding if the Reporter needs to know about the task.

Reply:
I've missed your comment and fixed it in this way:
```
class ImagelabReporterAdapter(Reporter):
    def __init__(
        self,
        data_issues: "DataIssues",
        imagelab: "Imagelab",
        task: str,
        verbosity: int = 1,
        include_description: bool = True,
        show_summary_score: bool = False,
    ):
        super().__init__(
            data_issues=data_issues,
            task=task,
            verbosity=verbosity,
            include_description=include_description,
            show_summary_score=show_summary_score,
        )
        self.imagelab = imagelab
```
but I'll reconsider your remark, especially that `Reporter` itself could be impacted by regression.